pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -189,73 +191,60 @@ int submit(long requestId, DispatchableSubPlan dispatchableSubPlan, long timeout
     return toResultTable(resultDataBlocks, dispatchableSubPlan.getQueryResultFields(),
         dispatchableSubPlan.getQueryStageList().get(0).getPlanFragment().getFragmentRoot().getDataSchema());
   }
 
-  private static List<DataBlock> reduceMailboxReceive(MailboxReceiveOperator mailboxReceiveOperator, long timeoutMs,
-      @Nullable Map<Integer, ExecutionStatsAggregator> executionStatsAggregatorMap,
-      DispatchableSubPlan dispatchableSubPlan,
-      OpChainStats stats) {
-    List<DataBlock> resultDataBlocks = new ArrayList<>();
-    TransferableBlock transferableBlock;
-    long timeoutWatermark = System.nanoTime() + timeoutMs * 1_000_000L;
-    while (System.nanoTime() < timeoutWatermark) {
-      transferableBlock = mailboxReceiveOperator.nextBlock();
-      if (TransferableBlockUtils.isEndOfStream(transferableBlock) && transferableBlock.isErrorBlock()) {
-        // TODO: we only received bubble up error from the execution stage tree.
-        // TODO: query dispatch should also send cancel signal to the rest of the execution stage tree.
-        throw new RuntimeException(
-            "Received error query execution result block: " + transferableBlock.getDataBlock().getExceptions());
-      }
-      if (transferableBlock.isNoOpBlock()) {
-        continue;
-      } else if (transferableBlock.isEndOfStreamBlock()) {
-        if (executionStatsAggregatorMap != null) {
-          for (Map.Entry<String, OperatorStats> entry : stats.getOperatorStatsMap().entrySet()) {
-            LOGGER.info("Broker Query Execution Stats - OperatorId: {}, OperatorStats: {}", entry.getKey(),
-                OperatorUtils.operatorStatsToJson(entry.getValue()));
-            OperatorStats operatorStats = entry.getValue();
-            ExecutionStatsAggregator rootStatsAggregator = executionStatsAggregatorMap.get(0);
-            ExecutionStatsAggregator stageStatsAggregator = executionStatsAggregatorMap.get(operatorStats.getStageId());
-            rootStatsAggregator.aggregate(null, entry.getValue().getExecutionStats(), new HashMap<>());
-            if (stageStatsAggregator != null) {
-              if (dispatchableSubPlan != null) {
-                OperatorUtils.recordTableName(operatorStats,
-                    dispatchableSubPlan.getQueryStageList().get(operatorStats.getStageId()));
-              }
-              stageStatsAggregator.aggregate(null, entry.getValue().getExecutionStats(), new HashMap<>());
-            }
+  private static void collectStats(DispatchableSubPlan dispatchableSubPlan, @Nullable OpChainStats opChainStats,
+      @Nullable Map<Integer, ExecutionStatsAggregator> executionStatsAggregatorMap) {
+    if (executionStatsAggregatorMap != null && opChainStats != null) {
+      for (Map.Entry<String, OperatorStats> entry : opChainStats.getOperatorStatsMap().entrySet()) {
+        LOGGER.info("Broker Query Execution Stats - OperatorId: {}, OperatorStats: {}", entry.getKey(),
```

Comment:
This logging is too expensive. This will get called for each (stageId, worker) pair which would be very large for most prod scenarios.

Reply:
good catch, this is already the case actually so will fix